misk/src/main/kotlin/misk/web/interceptors/GunzipRequestBodyInterceptor.kt

```diff
@@ -0,0 +1,36 @@
+package misk.web.interceptors
+
+import misk.Action
+import misk.web.NetworkChain
+import misk.web.NetworkInterceptor
+import misk.web.WebConfig
+import okio.GzipSource
+import okio.buffer
+import org.eclipse.jetty.server.handler.gzip.GzipHandler
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/**
+ * Inflates a gzipped compressed request. The interceptor should be installed if the default
+ * [GzipHandler] for handling such requests was opted out via [WebConfig.gunzip].
+ */
+class GunzipRequestBodyInterceptor private constructor() : NetworkInterceptor {
+  override fun intercept(chain: NetworkChain) {
+    val contentEncoding = chain.httpCall.requestHeaders["Content-Encoding"]
+    if (contentEncoding?.lowercase() == "gzip") {
```

Comment:
Should we also check for `x-gzip`? (Just judging by the Mozilla [Content-Encoding page](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Encoding#directives), if the existing implementation doesn't check for this then feel free to disregard)

Reply:
https://www.eclipse.org/jetty/javadoc/jetty-9/org/eclipse/jetty/server/handler/gzip/GzipHandler.html

Afaict from the specs, the interceptor should only observe for `gzip` and/or presence of it in a comma separated list - which I should update to reflect that.